### PR TITLE
Update mlflow server handlers.py to set tracking/registry uri after calculating them

### PR DESCRIPTION
**Pull request by akshaya-a**
## Related Issues/PRs

 #9531 #9495 

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by serena-ruan — APPROVED** [review]
LGTM! Could you merge master and rerun the failing tests?

**Comment by akshaya-a** [comment]
@serena-ruan  I know i'm useless with git but i think I rebased successfully - are you expecting these failing tests to have been fixed or should I look into them?

**Comment by serena-ruan** [comment]
> @serena-ruan I know i'm useless with git but i think I rebased successfully - are you expecting these failing tests to have been fixed or should I look into them?

I think it's related, the test doesn't fail on master. Also I found that if you just run that single test it doesn't fail, it only fails when running together with other tests..

**Comment by akshaya-a** [comment]
> > @serena-ruan I know i'm useless with git but i think I rebased successfully - are you expecting these failing tests to have been fixed or should I look into them?
> 
> I think it's related, the test doesn't fail on master. Also I found that if you just run that single test it doesn't fail, it only fails when running together with other tests..

thanks, I can take a look tomorrow, but then it sounds like something is assuming the old behavior in the test fixtures

**Comment by akshaya-a** [comment]
Actually this is strange, the entire suite passes for me locally.. Still digging but let me know if you have an idea of direction to investigate

**Comment by serena-ruan** [comment]
> Actually this is strange, the entire suite passes for me locally.. Still digging but let me know if you have an idea of direction to investigate

That is weird.. Yes that entire suite passes locally, but if you triggers the whole pytest tests/ then it fails. From my check it looks like the registry_uri here https://github.com/mlflow/mlflow/actions/runs/6153499312/job/16697465273?pr=9575#step:12:3140 is not correct, it should be the same as the tracking_uri

**Comment by akshaya-a** [comment]
As an update, I still cannot repro with this command line below, and the folders run in alphabetical order. So I suppose one of the autologging or root-level test files is the culprit. I'll continue digging as i get time between meetings
 
(mlflowwsl) akannava@CPC-akann-P9TVV:~/mlflow$ pytest -x --ignore=tests/mleap --ignore=tests/paddle --ignore=tests/gateway --ignore=tests/transformers --ignore=tests/autologging --ignore=tests/catboost --ignore=tests/test_cli.py --ignore=tests/data --ignore=tests/test_doctor.py --continue-on-collection-errors tests/

File: tests/test_cli.py [hunk 1]
```diff
@@ -111,6 +111,9 @@ def test_tracking_uri_validation_sql_driver_uris(command):
         )
         assert result.exit_code == 0
         run_server_mock.assert_called()
+    # Clean up the global variables set by the server
+    mlflow.tracking._tracking_service.utils.set_tracking_uri(None)
+    mlflow.tracking._model_registry.utils.set_registry_uri(None)
```
Comment (harupy) [inline tests/test_cli.py:116]: can we use `mlflow.set_tracking_uri` and `mlflow.set_registry_uri`?

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by harupy** [comment]
Can this PR close #9531 and #9495?

**Comment by akshaya-a** [comment]
> Can this PR close #9531 and #9495?

yep!

**Pull request merged**